mocks/middlewares/operation.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { RequestHandler } from 'express';
+import { NullableValues, paths } from '../types';
+
+export const operation = (): RequestHandler => (_req, res, next): void => {
+  res.operation = <Path extends keyof paths, Method extends keyof paths[Path]>(_path: Path, _method: Method) => ({
+    status: <Code extends number>(code: Code) => ({
+      send: (response: NullableValues<OperationResponseSchema<paths[Path][Method], Code>>): void => {
+        res.status(code).json(response);
+      },
+    }),
+  });
+  next();
+};
+
+type OperationResponseSchema<T, Code extends number> = T extends {
+  responses: {
+    [Key in Code]: { schema: infer Schema }
+  },
+} ? Schema : never;
+
+type OperationResponseCode<T> = T extends { responses: { [Key in infer P]: any } } ? P & number : never;
+
+declare global {
+  namespace Express {
+    interface Response {
+      operation: <Path extends keyof paths, Method extends keyof paths[Path]>(path: Path, method: Method) => {
+        status: <Code extends OperationResponseCode<paths[Path][Method]>>(code: Code) => {
+          send: (response: NullableValues<OperationResponseSchema<paths[Path][Method], Code>>) => void;
+        },
+      };
+    }
+  }
+}
```

Comment:
This part of code might be hard to understand so I'll detail it:
The idea is to provide a mechanism to ensure that we send the same responses as expected in the swagger spec.

With the generated types from the swagger spec, we have all the types of the api: paths, responses, operations and definitions. With those types, we can implement a type-safe mechanism to ensure we send the same response types as the production api.

To achieve that, I patched the express `res` object just like you would use `res.status()` or `res.json()`.

The usage is clearer than the implementation:
```ts
res
  .operation('/current-user', 'get') // with the path and the method, infer available status codes for `.status()` function
  .status(200) // with status code, infer response type for `.send()` function
  .send(…);
```